# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsContentUtils.h
# Commit: f9cf0740eef2
# Full Hash: f9cf0740eef2fcae3e2f93afe49df1569bd89c03
# Author: Adam Vandolder <avandolder@mozilla.com>
# Date: 2024-05-15 20:24:18
# Regressor Bug: 1890428
# File Overlap Count: 1
# Description:
#   Bug 1890428 - Add serialization support for Declarative Shadow DOM. r=webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D206955
# ==============================================================================

diff -r 37e2a10a9022 -r f9cf0740eef2 dom/base/nsContentUtils.h
--- a/dom/base/nsContentUtils.h	Tue May 14 16:43:27 2024 +0000
+++ b/dom/base/nsContentUtils.h	Tue May 14 16:43:28 2024 +0000
@@ -190,6 +190,7 @@
 class OwningFileOrUSVStringOrFormData;
 class Selection;
 enum class ShadowRootMode : uint8_t;
+class ShadowRoot;
 struct StructuredSerializeOptions;
 class WorkerPrivate;
 enum class ElementCallbackType;
@@ -232,6 +233,8 @@
 
 enum class TreeKind : uint8_t { DOM, Flat };
 
+enum class SerializeShadowRoots : uint8_t { Yes, No };
+
 struct EventNameMapping {
   // This holds pointers to nsGkAtoms members, and is therefore safe as a
   // non-owning reference.
@@ -3108,8 +3111,13 @@
   /*
    * Serializes a HTML nsINode into its markup representation.
    */
-  static bool SerializeNodeToMarkup(nsINode* aRoot, bool aDescendentsOnly,
-                                    nsAString& aOut);
+  template <SerializeShadowRoots ShouldSerializeShadowRoots =
+                SerializeShadowRoots::No>
+  static bool SerializeNodeToMarkup(
+      nsINode* aRoot, bool aDescendantsOnly, nsAString& aOut,
+      bool aSerializableShadowRoots,
+      const mozilla::dom::Sequence<
+          mozilla::OwningNonNull<mozilla::dom::ShadowRoot>>& aShadowRoots);
 
   /*
    * Returns true iff the provided JSObject is a global, and its URI matches
@@ -3543,7 +3551,7 @@
   MOZ_CAN_RUN_SCRIPT_BOUNDARY
   static nsIContent* AttachDeclarativeShadowRoot(
       nsIContent* aHost, mozilla::dom::ShadowRootMode aMode, bool aIsClonable,
-      bool aDelegatesFocus);
+      bool aIsSerializable, bool aDelegatesFocus);
 
  private:
   static bool InitializeEventTable();